src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java

```diff
@@ -412,7 +414,169 @@ private void checkVendorSpecificCapsProviders(GetCapabilitiesTransformer tr) thr
+        // the style won't appear
+        assertEquals(lgStyleName, "");
+    }
+
+    @Test
+    public void testDisableDefaultLayerGroupStyle1_3() throws Exception {
+        // test that when defaultGroupStyleEnabled
+        // is set to false the default layerGroup style doesn't appears in
+        // getCapabilities resp if mode is not single nor opaque.
+        String layerGroupName = "aLayerGroup";
+        createLayerGroup(layerGroupName);
+
+        Capabilities_1_3_0_Transformer tr =
+                new Capabilities_1_3_0_Transformer(
+                        wmsConfig,
+                        baseUrl,
+                        wmsConfig.getAllowedMapFormats(),
+                        wmsConfig.getAvailableExtendedCapabilitiesProviders());
+        Document dom = WMSTestSupport.transform(req, tr);
+
+        // default case, the style should be present
+        NodeList nodeList = dom.getElementsByTagName("Style");
+        assertEquals(1, nodeList.getLength());
+        Element styleEl = (Element) nodeList.item(0);
+        String title = styleEl.getElementsByTagName("Title").item(0).getTextContent();
+        assertEquals("aLayerGroup style", title);
+        WMS wms =
+                new WMS(geosConfig) {
+                    @Override
+                    public boolean isDefaultGroupStyleEnabled() {
+                        return false;
+                    }
+                };
+        Capabilities_1_3_0_Transformer tr2 =
+                new Capabilities_1_3_0_Transformer(
+                        wms,
+                        baseUrl,
+                        wms.getAllowedMapFormats(),
+                        wms.getAvailableExtendedCapabilitiesProviders());
+        Document dom2 = WMSTestSupport.transform(req, tr2);
+        nodeList = dom2.getElementsByTagName("Style");
+        // the style won't appear
+        assertEquals(0, nodeList.getLength());
+    }
+
+    @Test
+    public void testDefaultLayerGroupStyle1_1ModeSingle() throws Exception {
+        // test that when defaultGroupStyleEnabled
+        // is set to false the default layerGroup style appears in
+        // getCapabilities resp if mode is single.
+        String layerGroupName = "aLayerGroup";
+        createLayerGroup(layerGroupName, LayerGroupInfo.Mode.SINGLE);
+        WMS wms =
+                new WMS(geosConfig) {
+                    @Override
+                    public boolean isDefaultGroupStyleEnabled() {
+                        return false;
+                    }
+                };
+        GetCapabilitiesTransformer tr2 =
+                new GetCapabilitiesTransformer(wms, baseUrl, mapFormats, legendFormats, null);
+        Document dom2 = WMSTestSupport.transform(req, tr2);
+        String lgStyleTitle =
+                XPATH.evaluate("/WMT_MS_Capabilities/Capability/Layer/Layer/Style/Title", dom2);
+        // the style won't appear
```

Comment:
These comments say the style won't appear.. .but it does.

Reply:
fixed